File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        

        int i = 0;
        int j = 1;
        int ans = 0;

        while(j<prices.size())
        {
            int diff = prices[j] - prices[i];
            ans = max(ans ,diff);

            if(diff < 0)
            {
                i=j;
            }

            j++;

        }
        return ans;

    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {

        const set = new Set()
        let ans = 0;

        for(let i=0;i<nums.length; i++){
            set.add(nums[i])
        }

        for(let i of nums){
            
            if(!set.has(i-1)){
                let j=i;
                let maxi=0;
                while(set.has(j)){
                    maxi++
                    j++
                }
                ans = Math.max(ans,maxi)
            }

        }


        return ans

    }
}
